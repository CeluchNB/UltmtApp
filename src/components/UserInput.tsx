import React from 'react'
import { StyleSheet } from 'react-native'
import { TextInput } from 'react-native-paper'
import { useColors } from '../hooks/useColors'

interface UserInputProps {
    placeholder: string
    isPassword?: boolean
    onChangeText?: (...event: any[]) => void
    value?: string
}

const UserInput: React.FC<UserInputProps> = props => {
    const { colors } = useColors()

    const styles = StyleSheet.create({
        inputStyle: {
            backgroundColor: colors.primary,
            marginTop: 20,
            marginStart: 50,
            marginEnd: 50,
            placeholderTextColor: colors.secondary,
        },
    })

    return (
        <TextInput
            style={styles.inputStyle}
            placeholder={props.placeholder}
            mode="outlined"
            outlineColor={colors.textSecondary}
            activeOutlineColor={colors.textPrimary}
            secureTextEntry={props.isPassword}
            onChangeText={props.onChangeText}
            value={props.value}
            theme={{ colors: { text: colors.textPrimary } }}
        />
    )
}

export default UserInput
