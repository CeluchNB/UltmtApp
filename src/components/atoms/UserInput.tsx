import React from 'react'
import { TextInput } from 'react-native-paper'
import { useColors } from '../../hooks'
import { StyleProp, StyleSheet, ViewStyle } from 'react-native'

interface UserInputProps {
    placeholder: string
    isPassword?: boolean
    onChangeText?: (...event: any[]) => void
    value?: string
    style?: StyleProp<ViewStyle>
    rightIcon?: boolean
    onRightPress?: () => {}
    keyboardType?: 'default' | 'number-pad'
}

const UserInput: React.FC<UserInputProps> = ({
    placeholder,
    isPassword,
    onChangeText,
    value,
    style,
    rightIcon,
    onRightPress,
    keyboardType = 'default',
}) => {
    const { colors } = useColors()

    const styles = StyleSheet.create({
        inputStyle: {
            backgroundColor: colors.primary,
        },
        rightButton: {
            color: colors.textPrimary,
        },
    })

    return (
        <TextInput
            style={[styles.inputStyle, style]}
            placeholder={placeholder}
            placeholderTextColor={colors.secondary}
            textColor={colors.textPrimary}
            mode="outlined"
            outlineColor={colors.textSecondary}
            activeOutlineColor={colors.textPrimary}
            secureTextEntry={isPassword}
            onChangeText={onChangeText}
            value={value}
            keyboardType={keyboardType}
            autoCapitalize="none"
            right={
                rightIcon && (
                    <TextInput.Icon
                        icon="eye"
                        iconColor={colors.textPrimary}
                        onPressIn={onRightPress}
                        onPressOut={onRightPress}
                        testID="right-button"
                    />
                )
            }
            theme={{
                colors: {
                    text: colors.textPrimary,
                    placeholder: colors.secondary,
                },
            }}
        />
    )
}

export default UserInput
