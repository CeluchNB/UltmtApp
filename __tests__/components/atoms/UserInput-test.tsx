import * as React from 'react'
import UserInput from '../../../src/components/atoms/UserInput'
import { render } from '@testing-library/react-native'
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

it('test snapshot', () => {
    const snapshot = render(<UserInput placeholder="Placeholder" />).toJSON()
    expect(snapshot).toMatchSnapshot()
})

it('test with right icon', () => {
    const snapshot = render(
        <UserInput placeholder="Placeholder" rightIcon={true} />,
    ).toJSON()

    expect(snapshot).toMatchSnapshot()
})
