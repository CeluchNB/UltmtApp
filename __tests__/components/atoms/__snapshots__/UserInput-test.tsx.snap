// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<View
  style={{}}
>
  <View
    pointerEvents="none"
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 6,
        },
        {
          "backgroundColor": "#303030",
          "borderColor": "#7AC6FF",
          "borderRadius": 4,
          "borderWidth": 1,
        },
        undefined,
      ]
    }
    testID="text-input-outline"
  />
  <View>
    <View
      style={
        [
          {
            "paddingBottom": 0,
          },
          {
            "minHeight": 56,
            "paddingTop": 8,
          },
        ]
      }
    >
      <TextInput
        accessibilityLabel="Placeholder"
        autoCapitalize="none"
        cursorColor="#ffd600"
        editable={true}
        keyboardType="default"
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        placeholderTextColor="#ffffff"
        selectionColor="#ffd600"
        style={
          [
            {
              "margin": 0,
              "zIndex": 1,
            },
            {
              "height": 48,
            },
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "color": "#ffd600",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "paddingHorizontal": 16,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
            undefined,
          ]
        }
        testID="text-input-outlined"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;

exports[`test with right icon 1`] = `
<View
  style={{}}
>
  <View
    pointerEvents="none"
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 6,
        },
        {
          "backgroundColor": "#303030",
          "borderColor": "#7AC6FF",
          "borderRadius": 4,
          "borderWidth": 1,
        },
        undefined,
      ]
    }
    testID="text-input-outline"
  />
  <View>
    <View
      style={
        [
          {
            "paddingBottom": 0,
          },
          {
            "minHeight": 56,
            "paddingTop": 8,
          },
        ]
      }
    >
      <TextInput
        accessibilityLabel="Placeholder"
        autoCapitalize="none"
        cursorColor="#ffd600"
        editable={true}
        keyboardType="default"
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        placeholderTextColor="#ffffff"
        selectionColor="#ffd600"
        style={
          [
            {
              "margin": 0,
              "zIndex": 1,
            },
            {
              "height": 48,
            },
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "color": "#ffd600",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "paddingHorizontal": 16,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            {
              "marginRight": 56,
              "paddingRight": 0,
            },
            undefined,
          ]
        }
        testID="text-input-outlined"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "position": "absolute",
            "width": 24,
          },
          {
            "right": 16,
            "top": 19,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "height": 40,
            "margin": 0,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": 40,
          }
        }
        testID="right-button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderWidth": 0,
              "elevation": 0,
              "flex": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="right-button-container"
        >
          <View
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityTraits="button"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                [
                  {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                  {
                    "borderRadius": 20,
                  },
                ],
              ]
            }
            testID="right-button"
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                [
                  {
                    "color": "#ffd600",
                    "fontSize": 24,
                  },
                  [
                    {
                      "lineHeight": 24,
                      "transform": [
                        {
                          "scaleX": 1,
                        },
                      ],
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                  ],
                  {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              ó°ˆˆ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
